control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -376,6 +376,93 @@ impl ResourcePublishing for OperationGuardArc<VolumeSpec> {
 
         self.complete_update(registry, result, spec_clone).await
     }
+
+    async fn republish(
+        &mut self,
+        registry: &Registry,
+        request: &Self::Publish,
+    ) -> Result<Self::PublishOutput, SvcError> {
+        let specs = registry.specs();
+        let state = registry.get_volume_state(&request.uuid).await?;
+
+        // If the volume is not published then it should issue publish call rather than republish.
+        if state.target.is_none() {
+            return Err(SvcError::VolumeNotPublished {
+                vol_id: state.uuid.to_string(),
+            });
+        }
+
+        let nexus_node = get_volume_target_node(registry, &state, request, true).await?;
+        let newer_nexus_id = NexusId::new();
+
+        let operation =
+            VolumeOperation::Republish((nexus_node.clone(), newer_nexus_id.clone(), request.share));
+        let spec_clone = self.start_update(registry, &state, operation).await?;
+
+        let volume_target = spec_clone.target.as_ref().expect("already validated");
+        let mut older_nexus = specs.nexus(volume_target.nexus()).await?;
+        let older_nexus_id = older_nexus.lock().clone().uuid;
+        // shutdown the older nexus before newer nexus creation.
+        older_nexus
+            .shutdown(
+                registry,
+                &ShutdownNexus {
+                    uuid: older_nexus_id,
+                },
+            )
+            .await?;
+
+        // Create a Nexus on the requested or auto-selected node
+        let result = specs
+            .volume_create_nexus(registry, &nexus_node, &newer_nexus_id, &spec_clone)
+            .await;
+
+        let (mut nexus, nexus_state) = self
+            .validate_update_step(registry, result, &spec_clone)
+            .await?;
+
+        let (volume_id, last_nexus_id) = {
+            let volume_spec = self.lock();
+            (volume_spec.uuid.clone(), volume_spec.last_nexus_id.clone())
+        };
+
+        // Share the Nexus if it was requested
```

Comment:
I think republish should always imply re-sharing, so make it a requirement?

Reply:
Made it a requiement